File: app/src/main/java/com/example/drizzle/grappus_assignment/MainActivity.java
package com.example.drizzle.grappus_assignment;

import android.content.Intent;
import android.os.Bundle;
import android.os.Handler;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;
import android.text.Html;
import android.view.View;
import android.view.animation.Animation;
import android.view.animation.AnimationUtils;
import android.widget.Button;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.TextView;

import com.matthewtamlin.sliding_intro_screen_library.indicators.DotIndicator;

public class MainActivity extends AppCompatActivity {
    private ImageView logo_iv;
    private TextView tv_head_text,tv_sub_heading_text,tv_term_cond;
    private Button btn_skip, btn_create_profile;
    LinearLayout bottom_layout;
    DotIndicator indicator;
    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        logo_iv = findViewById(R.id.iv_logo);
        tv_head_text = findViewById(R.id.tv_head_text);
        tv_sub_heading_text = findViewById(R.id.tv_sub_heading_text);
        btn_skip = findViewById(R.id.btn_skip);
        btn_create_profile = findViewById(R.id.btn_create_profile);
        bottom_layout = findViewById(R.id.bottom_layout);
        indicator = findViewById(R.id.indicator);
        tv_term_cond = findViewById(R.id.tv_term_cond);
        tv_term_cond.setText(Html.fromHtml(String.format(getString(R.string.by_using_this_service_i_accept_athleteminder_s_term_of_service))));
        init();

    }

    private void init() {
      final  Animation shake_animation = AnimationUtils.loadAnimation(getApplicationContext(), R.anim.shake);

        Animation animZoomOut = AnimationUtils.loadAnimation(getApplicationContext(),R.anim.zoom_out);
        logo_iv.startAnimation(animZoomOut);
        new Handler().postDelayed(new Runnable() {

            @Override
            public void run() {
                tv_head_text.setVisibility(View.VISIBLE);
            }
        }, 1000);
        new Handler().postDelayed(new Runnable() {

            @Override
            public void run() {
                tv_sub_heading_text.setVisibility(View.VISIBLE);
            }
        }, 1300);
        new Handler().postDelayed(new Runnable() {

            @Override
            public void run() {
                btn_create_profile.setVisibility(View.VISIBLE);
                btn_create_profile.startAnimation(shake_animation);
            }
        }, 1600);
        new Handler().postDelayed(new Runnable() {

            @Override
            public void run() {
                btn_skip.setVisibility(View.VISIBLE);
                btn_skip.startAnimation(shake_animation);
            }
        }, 1800);
        new Handler().postDelayed(new Runnable() {

            @Override
            public void run() {
                bottom_layout.setVisibility(View.VISIBLE);
                indicator.setVisibility(View.VISIBLE);
                btn_skip.startAnimation(shake_animation);
            }
        }, 2000);

        btn_skip.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent =  new Intent(MainActivity.this, HomeActivity.class);
                startActivity(intent);

            }
        });
    }

}

File: app/src/main/java/com/example/drizzle/grappus_assignment/MainActivity.kt
package com.example.drizzle.grappus_assignment;

import android.content.Intent
import android.os.Bundle
import android.os.Handler
import android.support.v7.app.AppCompatActivity
import android.text.Html
import android.view.animation.AnimationUtils
import com.facebook.rebound.*
import kotlinx.android.synthetic.main.activity_main.*



class MainActivity : AppCompatActivity() {

    private companion object {
        private val TENSION = 40.0
        private val FRICTION = 6.0
        private val START_SCALE = 0.0
        private val END_SCALE = 1.0

        private val START_LOGO_ANIM: Long = 500
        private val START_HEADING_ANIM: Long = 2000
        private val START_BUTTON_ANIM: Long = 2500
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        setContentView(R.layout.activity_main)

        tv_term_cond.text = Html.fromHtml(String.format(getString(R.string.by_using_this_service_i_accept_athleteminder_s_term_of_service)))

        btn_skip.setOnClickListener {
            val intent = Intent(this@MainActivity, HomeActivity::class.java)
            startActivity(intent)
        }

        Handler().postDelayed({
            animateLogoRebound()
        }, START_LOGO_ANIM)

        Handler().postDelayed({
            animateHeadingAndSubHeading()
        }, START_HEADING_ANIM)

        Handler().postDelayed({
            animateButton()
        }, START_BUTTON_ANIM)
    }

    private fun animateLogoRebound() {
        val springSystem = SpringSystem.create()
        val spring = springSystem.createSpring()
        spring.springConfig = SpringConfig(TENSION, FRICTION)
        spring.endValue = 1.0

        spring.addListener(object : SimpleSpringListener() {

            override fun onSpringUpdate(spring: Spring) {
                // You can observe the updates in the spring
                // state by asking its current value in onSpringUpdate.
                val value = spring.currentValue.toFloat()
                val mappedValue = SpringUtil.mapValueFromRangeToRange(
                        spring.currentValue, 0.0, 1.0, START_SCALE, END_SCALE).toFloat()
                val scale = 1f - value * 0.5f
                iv_logo.scaleX = mappedValue
                iv_logo.scaleY = mappedValue
            }

            override fun onSpringAtRest(spring: Spring) {

            }

            override fun onSpringActivate(spring: Spring) {

            }

            override fun onSpringEndStateChange(spring: Spring) {

            }
        })
    }

    private fun animateHeadingAndSubHeading() {
        val animationFadeIn = AnimationUtils.loadAnimation(this@MainActivity, R.anim.fade_in)
        animationFadeIn.fillAfter = true
        tv_head_text.startAnimation(animationFadeIn)
        tv_sub_heading_text.startAnimation(animationFadeIn)
    }

    private fun animateButton() {
        val animationFadeIn = AnimationUtils.loadAnimation(this@MainActivity, R.anim.fade_in)
        animationFadeIn.fillAfter = true
        btn_create_profile.startAnimation(animationFadeIn)

        Handler().postDelayed({
            val animationFadeInInner = AnimationUtils.loadAnimation(this@MainActivity, R.anim.fade_in)
            animationFadeInInner.fillAfter = true
            btn_skip.startAnimation(animationFadeInInner)
            tv_term_cond.startAnimation(animationFadeInInner)
            indicator.startAnimation(animationFadeInInner)
        }, 250)

    }
}
